src/compiler/parser.ts

```diff
@@ -5139,8 +5139,26 @@ namespace Parser {
     }
 
     function nextTokenIsIdentifierOnSameLine() {
+        const isTypeKeyword = token() === SyntaxKind.TypeKeyword;
         nextToken();
-        return !scanner.hasPrecedingLineBreak() && isIdentifier();
+
+        if (scanner.hasPrecedingLineBreak()) {
+            return false;
+        }
+
+        if (!isTypeKeyword) {
+            return isIdentifier();
+        }
+
+        switch (token()) {
+            case SyntaxKind.AsKeyword:

```

Comment:
It really feels like this function should have been left as-is, and the look-ahead should have been done after calling this function specifically for the `type` keyword.

~~It also might be the case that you don't actually need to special-case `as` and `satisfies`, you can just do an unconditional look-ahead. But I might have to think about that.~~